sdk/python/lib/pulumi/runtime/settings.py

```diff
@@ -98,6 +95,73 @@ def __init__(
         else:
             self.engine = None
 
+    def __repr__(self):
+        return f"Settings: {self.engine}:{self.monitor}:{self.project}:{self.stack}"
+
+    @property
+    def engine(self):
+        return self._engine.get()
+
+    @engine.setter
+    def engine(self, v: Optional[Union[engine_pb2_grpc.EngineStub, Any]]):
+        return self._engine.set(v).var.get()
+
+    @property
+    def monitor(self):
+        return self._monitor.get()
+
+    @monitor.setter
+    def monitor(self, v: Optional[Union[resource_pb2_grpc.ResourceMonitorStub, Any]]):
+        return self._monitor.set(v).var.get()
+
+    @property
+    def project(self):
+        return self._project.get()
+
+    @project.setter
+    def project(self, v: Optional[str]):
+        return self._project.set(v).var.get()
+
+    @property
+    def stack(self):
+        return self._stack.get()
+
+    @stack.setter
+    def stack(self, v: Optional[str]):
+        return self._stack.set(v).var.get()
+
+    @property
+    def dry_run(self):
+        return self._dry_run.get()
+
+    @dry_run.setter
+    def dry_run(self, v: Optional[bool]):
+        return self._dry_run.set(v).var.get()
+
+    @property
+    def test_mode_enabled(self):
+        return self._test_mode_enabled.get()
+
+    @test_mode_enabled.setter
+    def test_mode_enabled(self, v: Optional[bool]):
+        return self._test_mode_enabled.set(v).var.get()
+
+    @property
+    def legacy_apply_enabled(self):
+        return self._legacy_apply_enabled.get()
+
+    @legacy_apply_enabled.setter
+    def legacy_apply_enabled(self, v: Optional[bool]):
+        return self._legacy_apply_enabled.set(v).var.get()
+
+    @property
+    def feature_support(self):
+        return self._feature_support.get()
+
+    @feature_support.setter
+    def feature_support(self, v: Optional[dict]):
+        return self._feature_support.set(v).var.get()
```

Comment:
This will need to be a bit smarter.  I think this will need to be an object which overrides `__getitem__` to adapt the dict interface to a ContextVar'd dict